### PR TITLE
Add problem package and statement to problems endpoint

### PR DESCRIPTION
**Pull request by niemela**
Add the problem package and statement to the problems endpoint.

Also, merge the file reference types (IMAGE, VIDEO, ARCHIVE, STREAM) into a single file reference type (FILE). Note that FILE was already used, but not defined.

### CODE REVIEW

**Review by deboer-tim — COMMENTED** [review]
I wasn't expecting the merge of file reference types. There is a slight drawback in that STREAMs are now FILEs, but that was inherent in the file reference type name anyway.

I am concerned with the new requirement for videos (team and reactions) & streams to have a width and height. We haven't done this before, and the latter requires us to either hardcode it or connect to every stream - would like to discuss on the next call.

**Comment by niemela** [comment]
> I am concerned with the new requirement for videos (team and reactions) & streams to have a width and height. 

I would argue that this is not new. The old text said:
`Width of the image, video or stream in pixels. Should not be used for ARCHIVE.`

> We haven't done this before, and the latter requires us to either hardcode it or connect to every stream - would like to discuss on the next call.

I would have no trouble changing to:
`Width of the image. Required for files with mime type image/*.` (and same for height).
I only added the rest to *not* make a change.

**Comment by deboer-tim** [comment]
> I would have no trouble changing to: `Width of the image. Required for files with mime type image/*.` (and same for height). I only added the rest to _not_ make a change.

Before it was "Nullable: no for **`IMAGE`**", now it's "Nullable: depends" and "Required for files with mime type image/* and video/*", which sounded like a change to me. I'd be good if we remove the "and video/*" as you suggested, or we can discuss.

**Comment by niemela** [comment]
> I'd be good if we remove the "and video/*" as you suggested

Done.

File: Contest_API.md [hunk 1]
```diff
@@ -900,6 +900,8 @@ JSON elements of problem objects:
 | time\_limit       | number  | no        | no        | Time limit in seconds per test data set (i.e. per single run). Should be an integer multiple of `0.001`.
 | test\_data\_count | integer | yes       | no        | Number of test data sets.
 | max_score         | number  | no        | no        | Maximum expected score, although teams may score higher in some cases. Typically used to indicate scoreboard cell color in scoring contests. Required iff contest:scoreboard_type is `score`.
+| problem_package   | array of FILE  | no | yes    | [Problem package](https://www.kattis.com/problem-package-format/). Expected mime type is application/zip. Only exactly one archive is allowed. Not expected to actually contain href for package during the contest, but used for configuration and archiving.
+| problem_statememt | array of FILE | no | yes    | Problem statemet. Expected mime type is application/pdf.
```
Comment (ghamerly) [inline Contest_API.md:904]:
```suggestion
| problem_statement | array of FILE | no | yes    | Problem statement. Expected mime type is application/pdf. 
```

**Review by ghamerly — DISMISSED** [review]
Looks okay to me, with the one suggestion that I made (typo fixes).

File: Contest_API.md [hunk 2]
```diff
@@ -900,6 +900,8 @@ JSON elements of problem objects:
 | time\_limit       | number  | no        | no        | Time limit in seconds per test data set (i.e. per single run). Should be an integer multiple of `0.001`.
 | test\_data\_count | integer | yes       | no        | Number of test data sets.
 | max_score         | number  | no        | no        | Maximum expected score, although teams may score higher in some cases. Typically used to indicate scoreboard cell color in scoring contests. Required iff contest:scoreboard_type is `score`.
+| problem_package   | array of FILE  | no | yes    | [Problem package](https://www.kattis.com/problem-package-format/). Expected mime type is application/zip. Only exactly one archive is allowed. Not expected to actually contain href for package during the contest, but used for configuration and archiving.
```
Comment (nickygerritsen) [inline Contest_API.md:903]: I get what you want to say with the sentence, but as a reader I got questions. Like "it's not expected during the contest, but how about before? Or after?" and "When **is** it expected?"

Comment (niemela) [reply Contest_API.md:903]: For configuration and archiving?

Do you have a suggestion for how to explain it better?

Comment (nickygerritsen) [reply Contest_API.md:903]: Not really. Maybe “for package in live systems” but that also is meh

Comment (eldering) [inline Contest_API.md:903]: Did we ever decide/clarify whether a problem package zip contains the files inside a directory of the problem name or in the root of the zip? I guess that's something that has to be clarified in the problem-package-format repo, but I think it's relevant here, as this is meant for potentially build automated tooling on top of that file format.

File: Contest_API.md [hunk 3]
```diff
@@ -1019,20 +1021,20 @@ organizations (i.e. return any organization\_ids).
...
-| logo               | array of IMAGE | no        | yes       | Logo of the organization. A server must provide logos of size 56x56 and 160x160 but may provide other sizes as well.
+| Name               | Type          | Required? | Nullable? | Description
+| :----------------- | :------------ | :-------- | :-------- | :----------
+| id                 | ID            | yes       | no        | Identifier of the organization.
+| icpc\_id           | string        | no        | yes       | External identifier from ICPC CMS.
+| name               | string        | yes       | no        | Short display name of the organization.
+| formal\_name       | string        | no        | yes       | Full organization name if too long for normal display purposes.
+| country            | string        | no        | yes       | [ISO 3166-1 alpha-3 code](https://en.wikipedia.org/wiki/ISO_3166-1_alpha-3) of the organization's country.
+| country_flag       | array of FILE | no        | yes       | Flag of the country. A server is recommended to provide flags of size around 56x56 and 160x160. Only allowed mime types are image/*.
```
Comment (eldering) [inline Contest_API.md:1031]: I'm assuming that here and below the intention is that this is still restricted to the 3 image mime types we support: png, jpg and svg, right? That's still mentioned above but is now a little less explicit, but I think that's ok.

File: Contest_API.md (hunk 2 above)
Comment (niemela) [reply Contest_API.md:903]: I think what I wrote is better.

Comment (niemela) [reply Contest_API.md:903]: > Did we ever decide/clarify whether a problem package zip contains the files inside a directory of the problem name or in the root of the zip?

I don't remember.

I think we should say that the directory must be included.

> I guess that's something that has to be clarified in the problem-package-format repo, 

Yes.

> but I think it's relevant here, as this is meant for potentially build automated tooling on top of that file format.

Sure it's relevant, but the correct place to document all the details about the problem package format is in the problem package format specification. 

In summary, I don't think we should change the current wording here, but we should update the problem package format to clarify this.

File: Contest_API.md (hunk 3 above)
Comment (niemela) [reply Contest_API.md:1031]: > I'm assuming that here and below the intention is that this is still restricted to the 3 image mime types we support: png, jpg and svg, right? 

Yes.

> That's still mentioned above but is now a little less explicit, but I think that's ok.

I agree.

**Review by nickygerritsen — APPROVED** [review]

**Review by ghamerly — APPROVED** [review]

**Pull request merged**